web/packages/teleport/src/components/ExternalAuditStorageCta/ExternalAuditStorageCta.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+import Box from 'design/Box';
+import * as Icons from 'design/Icon';
+import { ButtonPrimary, ButtonSecondary } from 'design/Button';
+import Flex from 'design/Flex';
+import Text from 'design/Text';
+
+import cfg from 'teleport/config';
+import { IntegrationKind } from 'teleport/services/integrations';
+import localStorage from 'teleport/services/localStorage';
+import useTeleport from 'teleport/useTeleport';
+
+import { CtaEvent } from 'teleport/services/userEvent';
+
+import { ButtonLockedFeature } from '../ButtonLockedFeature';
+
+export const Container = () => {
```

Comment:
In the new components we write, [we decided to move away from our old design pattern](https://gravitational.slack.com/archives/C02DQ1C2BMW/p1675213044004249) of having all logic handled in a `Container` which passes the props to the component.

We should move all this logic into the body of the `ExternalAuditStorageCta` component itself and scrap the `Container` wrapper, and use mocking in tests as needed to simulate the behaviour we want.

Reply:
+1, thanks for catching this